File: caesar.py
from sys import argv
import sys
import string

# Raise exception for no, or more than 1 argument.
if(len(argv) != 2):
	print 'Only one command line argument allowed.'
	exit(1)

"""
We unpack the argument vector so that 'key' holds the
(numeric) key used for implementing the cipher.
Since it is a string we convert it to an int.
"""
script, key = argv
key = int(key)
string = raw_input('Enter the string you want encrypted\n')
size = len(string)

"""
To maintain numbers as they are, and to maintain case,
we implement some if-constructs. After filtering out
the aplhabetic portion of the string, we change
the characters to their respective ASCII values.
Begin counting from 0, to 26. Add the key, using
modulo arithmetic in the process, and then get
it back upto ASCII values and print them.
"""
for i in range(0,size):
	# Check if alphabet.
	if(string[i].isalpha()):
		# Check if lowercase.
		if(string[i].islower()):

			temp = ord(string[i])
			temp -= ord('a')
			temp += key
			temp %= 26
			temp += ord('a')
			sys.stdout.write(chr(temp))

		else:

			temp = ord(string[i])
			temp -= ord('A')
			temp += key
			temp %= 26
			temp += ord('A')
			sys.stdout.write(chr(temp))
	else:

		sys.stdout.write(string[i])

print '\n'

exit(0)

File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{	

	// Raise exception for no, or more than 1 arguments.
	if(argc != 2)
	{
		printf("Exactly one numeric command line argument allowed.\n");
		return 1;
	}
	
	// Convert char to int.			
	int key = atoi(argv[1]);
	string input = GetString();
	int size = strlen(input);
	
	/* The following block of code first bring the alphabet down 
	 * to between 0 and 26. Then we add the key and use modulo
	 * 26 to wrap around 'z' and continue back from 'a'. Then we
	 * again get it up to its ASCII value and save it.
	*/
		
	for(int i = 0; i < size; i++)
	{
		if(isalpha(input[i]))
		{	
			if(islower(input[i]))
			{
				input[i] = input[i] - 'a';
				input[i] += key;
				input[i] %= 26;
				input[i] += 'a';
				//printf("%c", input[i]); // DEBUG POINT
			}
			else
			{
				input[i] = input[i] - 'A'; 
				input[i] += key;
				input[i] %= 26;
				input[i] += 'A';
			}
		}
	}
	
	// Print the ciphered text.
	for (int i = 0; i < size; i++)
	{
		printf("%c", input[i]);
	}
	
	printf("\n");
	
	return 0;
	
}
